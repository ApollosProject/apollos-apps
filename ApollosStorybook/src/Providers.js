import React from 'react';
import ApollosConfig from '@apollosproject/config';
import { Providers } from '@apollosproject/ui-kit';
import { MediaPlayerProvider } from '@apollosproject/ui-media-player';
import { LiveProvider } from '@apollosproject/ui-connected';
import ClientProvider, { client } from './client';
import customTheme, { customIcons } from './theme';

const AppProviders = (props) => (
  <ClientProvider>
    <MediaPlayerProvider>
        <LiveProvider>
          <Providers
            themeInput={customTheme}
            iconInput={customIcons}
            {...props}
          />
        </LiveProvider>
    </MediaPlayerProvider>
  </ClientProvider>
);

export default AppProviders;
