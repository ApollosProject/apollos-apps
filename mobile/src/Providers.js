import { nest } from 'recompose';

import ClientProvider from '/mobile/client';
import { ThemeProvider } from '/mobile/ui/theme';
import { WebBrowserProvider } from '/mobile/ui/WebBrowser';

export default nest(ClientProvider, ThemeProvider, WebBrowserProvider);
