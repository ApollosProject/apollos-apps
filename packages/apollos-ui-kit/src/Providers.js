import { nest } from 'recompose';

import { ThemeProvider } from './theme';
import { LayoutProvider } from './LayoutContext';

export default nest(ThemeProvider, LayoutProvider);
