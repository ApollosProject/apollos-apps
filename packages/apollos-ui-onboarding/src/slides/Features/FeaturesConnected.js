import React from 'react';
import { Query } from 'react-apollo';
import PropTypes from 'prop-types';

import GET_USER_FIRST_NAME from './getUserFirstName';
import Features from './Features';

const FeaturesConnected = ({ Component, ...props }) => (
  <Query query={GET_USER_FIRST_NAME}>
    {({
      loading,
      data: {
        currentUser: { profile: { firstName } = { campus: {} } } = {},
      } = {},
    }) => (
      <Component
        firstName={firstName}
        isLoading={loading}
        pressPrimaryEventName={'Features Completed'}
        {...props}
      />
    )}
  </Query>
);

FeaturesConnected.propTypes = {
  // Custom component to be rendered. Defaults to Features
  Component: PropTypes.oneOfType([
    PropTypes.node,
    PropTypes.func,
    PropTypes.object,
  ]),
};

FeaturesConnected.defaultProps = {
  Component: Features,
};

FeaturesConnected.displayName = 'FeaturesConnected';

export default FeaturesConnected;
