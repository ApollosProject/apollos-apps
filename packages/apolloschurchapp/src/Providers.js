import { nest } from 'recompose';

import ClientProvider from 'apolloschurchapp/src/client';
import { ThemeProvider } from 'apolloschurchapp/src/ui/theme';
import { WebBrowserProvider } from 'apolloschurchapp/src/ui/WebBrowser';

export default nest(ClientProvider, ThemeProvider, WebBrowserProvider);
