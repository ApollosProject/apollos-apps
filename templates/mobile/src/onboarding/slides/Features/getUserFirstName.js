import { gql } from '@apollo/client';

export default gql`
  query getUserFirstName {
    currentUser {
      id
      profile {
        id
        firstName
      }
    }
  }
`;
