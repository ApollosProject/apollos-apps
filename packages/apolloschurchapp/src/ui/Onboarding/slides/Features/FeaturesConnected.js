import React from 'react';
import { Query } from 'react-apollo';

import getUserFirstName from './getUserFirstName';
import Features from '.';

const AskNameConnected = (props) => (
  <Query query={getUserFirstName}>
    {({
      loading,
      data: {
        currentUser: { profile: { firstName } = { campus: {} } } = {},
      } = {},
    }) => <Features firstName={firstName} isLoading={loading} {...props} />}
  </Query>
);

AskNameConnected.displayName = AskNameConnected;

export default AskNameConnected;
