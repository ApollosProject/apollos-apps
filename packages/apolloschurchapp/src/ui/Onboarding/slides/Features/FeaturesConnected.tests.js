import React from 'react';
import renderer from 'react-test-renderer';
import wait from 'waait';

import Providers from 'apolloschurchapp/src/Providers';

import getUserFirstName from './getUserFirstName';
import FeaturesConnected from './FeaturesConnected';

describe('The Onboarding FeaturesConnected component', () => {
  it('renders with a firstName when logged in', async () => {
    const mock = {
      request: {
        query: getUserFirstName,
      },
      result: {
        data: {
          currentUser: {
            profile: {
              firstName: 'Marty',
            },
          },
        },
      },
    };
    const tree = renderer.create(
      <Providers mocks={[mock]}>
        <FeaturesConnected />
      </Providers>
    );
    await wait(0); // wait for response from graphql
    expect(tree).toMatchSnapshot();
  });
});
