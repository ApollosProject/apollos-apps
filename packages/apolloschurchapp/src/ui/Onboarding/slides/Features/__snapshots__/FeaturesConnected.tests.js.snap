// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`The Onboarding FeaturesConnected component renders with a firstName when logged in 1`] = `
<RCTSafeAreaView
  emulateUnlessSupported={true}
  onLayout={[Function]}
  style={
    Object {
      "bottom": 0,
      "left": 0,
      "position": "absolute",
      "right": 0,
      "top": 0,
    }
  }
>
  <View
    onLayout={[Function]}
    style={
      Object {
        "flex": 1,
      }
    }
  />
  <View
    style={
      Object {
        "bottom": 0,
        "left": 0,
        "position": "absolute",
        "right": 0,
        "top": 0,
      }
    }
  >
    <RCTScrollView
      contentContainerStyle={
        Object {
          "minHeight": "100%",
        }
      }
      overScrollMode="auto"
      style={
        Object {
          "flex": 1,
        }
      }
    >
      <View>
        <RCTSafeAreaView
          emulateUnlessSupported={true}
          forceInset={
            Object {
              "bottom": "always",
            }
          }
          style={
            Object {
              "marginBottom": 48,
              "paddingHorizontal": 16,
              "paddingTop": 16,
            }
          }
        >
          <View>
            <View
              style={
                Object {
                  "paddingBottom": 8,
                }
              }
            >
              <Text
                style={
                  Object {
                    "color": "#00676D",
                    "fontFamily": "InterUI-Black",
                    "fontSize": 36,
                    "lineHeight": 41.4,
                  }
                }
              >
                Hey friend!
              </Text>
            </View>
            <Text
              style={
                Object {
                  "color": "#505050",
                  "fontFamily": "InterUI-Medium",
                  "fontSize": 14,
                  "lineHeight": 20.16,
                }
              }
            >
              We'd like to help personalize your mobile experience so we can help you with every step on your journey.
            </Text>
          </View>
        </RCTSafeAreaView>
      </View>
    </RCTScrollView>
  </View>
</RCTSafeAreaView>
`;
